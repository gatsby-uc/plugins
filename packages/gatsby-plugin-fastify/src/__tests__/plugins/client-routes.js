describe(`Client Side Routes`, () => {
  it(`Should handle base route`, async () => {
    const response = await fastify.inject({
      url: "/app/",
      method: "GET",
    });

    expect(response.statusCode).toEqual(200);
    expect(response.payload).toMatchSnapshot();
  });

  it(`Should handle sub routes`, async () => {
    const responseBase = await fastify.inject({
      url: "/app/",
      method: "GET",
    });

    const response = await fastify.inject({
      url: "/app/subPath",
      method: "GET",
    });

    expect(response.statusCode).toEqual(200);
    expect(response.payload).toMatchSnapshot();
    expect(responseBase.payload).toEqual(response.payload);
  });
});
