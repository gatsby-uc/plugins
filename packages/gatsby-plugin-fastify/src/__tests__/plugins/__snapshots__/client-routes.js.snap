// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Client Side Routes Should handle base route 1`] = `
"<!DOCTYPE html>
<html lang="en">

<head>
  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <title>Gatsby Plugin Fastify - Client Side Route</title>
</head>

<body>
  <h1>Welcome to the CSR!</h1>
</body>

</html>"
`;

exports[`Client Side Routes Should handle sub routes 1`] = `
"<!DOCTYPE html>
<html lang="en">

<head>
  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <title>Gatsby Plugin Fastify - Client Side Route</title>
</head>

<body>
  <h1>Welcome to the CSR!</h1>
</body>

</html>"
`;
